storage/internal/benchmarks/client_pool.go

```diff
@@ -197,8 +191,8 @@ func initializeHTTPClient(ctx context.Context, writeBufferSize, readBufferSize i
 	return client, err
 }
 
-func initializeGRPCClient(ctx context.Context, writeBufferSize, readBufferSize int, connPoolSize int, useDefaults bool) (*storage.Client, error) {
-	if useDefaults {
+func initializeGRPCClient(ctx context.Context, p clientParams) (*storage.Client, error) {
+	if !p.customClient {
```

Comment:
Also a bit off topic, but the flow of the logic here is kind of confusing. How about something like:
```
var opts []option.ClientOption
if !p.CustomClient {
  // set custom client opts
}
clientMu.Lock()
// set env var and initialize client using opts...
clientMu.Unlock()
```

Same thing for HTTP.

Reply:
I did just this (I think) in my follow-up PR that adds JSON and removes the customClient var. I think we can leave this as-is for this PR and discuss the flow in that one.

Reply:
Okay sounds good!